packages/cdktf/lib/terraform-stack.ts

```diff
@@ -184,6 +228,64 @@ export class TerraformStack extends Construct {
 
     return resolve(this, tf);
   }
+
+  public registerOutgoingCrossStackReference(identifier: string) {
```

Comment:
Can register accidentally be called with the same identifier for different references?

Reply:
Yes it can, even intentionally, e.g. if you take the VPC example and reference the VPC in two different application stacks. This would call the function twice, creating only one (since caching is in place) output. 

This is no problem though, we just need the data in an output, there is no need to have one output per referencing stack.

Reply:
Ah, got it